scheduler.go

```diff
@@ -298,7 +298,12 @@ func (s *scheduler) selectExecJobsOutForRescheduling(id uuid.UUID) {
 		// so we don't need to reschedule it.
 		return
 	}
-	j.lastScheduledRun = j.nextScheduled
+	if len(j.nextScheduled) > 0 {
+		// always grab the last element in the slice as that is the furthest
+		// out in the future and the time from which we want to calculate
+		// the subsequent next run time.
+		j.lastScheduledRun = j.nextScheduled[len(j.nextScheduled)-1]
+	}
```

Comment:
There's an assumption here that the jobs in `j.nextScheduled` are sorted. I understand that the code may be written in such a way that sorting is not needed, but it _might_ be a good idea to sort `nextScheduled` and then get the last element in the slice. What do you think?

Reply:
Definitely a safe thing to do 👍